webhooks/framework.go

```diff
@@ -0,0 +1,55 @@
+package webhooks
+
+import (
+	"fmt"
+	jwt "github.com/dgrijalva/jwt-go"
+	"github.com/rancher/webhook-service/drivers"
+)
+
+// ConstructPayload constructs jwt url
+func ConstructPayload(input map[string]interface{}) (map[string]interface{}, error) {
+	if _, ok := input["driver"].(string); !ok {
+		return nil, fmt.Errorf("Driver of type string not provided\n")
+	}
+	driverId := input["driver"].(string)
+	driver := drivers.GetDriver(driverId)
+	if driver == nil {
+		return nil, fmt.Errorf("Driver %s is not registered\n", driverId)
+	}
+	payload, err := driver.ConstructPayload(input)
```

Comment:
@cjellick sorry I didn't get that, I thought this is generic. The reason being it only looks up the driver and calls it. This is what I thought the flow would be:
1. service/ConstructPayload(w http..., r...) only unmarshals json from request body to map and calls webhooks/ConstructPayload that is also same for all drivers
2. webhooks/ConstructPayload(input map[string]interface{}) (map..., error) will just look up the driver and call that driver's constructPayload.
3. drivers/ConstructPayload is called from step 2, which is driver specific and for ServiceScaler is only creating jwt
Please let me know if I've misunderstood this